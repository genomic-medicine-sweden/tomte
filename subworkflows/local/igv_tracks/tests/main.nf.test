nextflow_workflow {

    name "Test Workflow IGV_TRACKS"
    script "../main.nf"
    workflow "IGV_TRACKS"
    config "./nextflow.config"
    tag "igv_tracks"

    setup {
        run("GUNZIP", alias: "GUNZIP_FASTA") {
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'fasta' ],
                    file("${projectDir}/test_data/grch37_chr21.fa.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("GUNZIP", alias: "GUNZIP_GTF") {
            script "../../../../modules/nf-core/gunzip/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'gtf' ],
                    file("${projectDir}/test_data/grch37_chr21.gtf.gz", checkIfExists: true)
                ])
                """
            }
        }

        run("SAMTOOLS_FAIDX") {
            script "../../../../modules/nf-core/samtools/faidx/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = [[],[]]
                """
            }
        }

        run("GET_CHROM_SIZES") {
            script "../../../../modules/local/get_chrom_sizes"
            process {
                """
                input[0] = SAMTOOLS_FAIDX.out.fai
                """
            }
        }

        run("STAR_GENOMEGENERATE") {
            script "../../../../modules/nf-core/star/genomegenerate/main.nf"
            process {
                """
                input[0] = GUNZIP_FASTA.out.gunzip
                input[1] = GUNZIP_GTF.out.gunzip
                """
            }
        }

        run("FASTP") {
            script "../../../../modules/nf-core/fastp/main.nf"
            process {
                """
                input[0] = Channel.of([[case:'finequagga', id:'ACC5963A1', sample:'ACC5963A1', strandedness:'reverse', sex:'NA', single_end:false, fq_pairs:1, is_fastq:true], 
                    [file("${projectDir}/test_data/ACC5963A1_sub_1.fastq.gz", checkIfExists: true),
                    file("${projectDir}/test_data/ACC5963A1_sub_2.fastq.gz", checkIfExists: true)]
                    ])
                input[1] = []
                input[2] = false
                input[3] = false
                input[4] = false
                """
            }
        }

        run("STAR_ALIGN") {
            script "../../../../modules/nf-core/star/align/main.nf"
            process {
                """
                input[0] = FASTP.out.reads
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = GUNZIP_GTF.out.gunzip
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }
    }

    test("Should run without failures") {

        when {
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = STAR_ALIGN.out.wig
                input[1] = GET_CHROM_SIZES.out.sizes.collect()
                input[2] = STAR_ALIGN.out.spl_junc_tab
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
