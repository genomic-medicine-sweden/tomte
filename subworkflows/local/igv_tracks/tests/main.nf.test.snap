{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "case": "finequagga",
                            "id": "ACC5963A1",
                            "sample": "ACC5963A1",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        },
                        "ACC5963A1.bw:md5,bd693d63017f2d313aa50d8befdf339a"
                    ]
                ],
                "1": [
                    [
                        {
                            "case": "finequagga",
                            "id": "ACC5963A1",
                            "sample": "ACC5963A1",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        },
                        "ACC5963A1_junction.bed.gz:md5,af36db589fe4bf79ab49944ef10d0891",
                        "ACC5963A1_junction.bed.gz.tbi:md5,30d9506483e03a1bc0fba299faba1a65"
                    ]
                ],
                "2": [
                    "versions.yml:md5,5abec6f51f30415625cf8ce28c698a72",
                    "versions.yml:md5,f4eb74ccf752488d150f4be9e1eb2142",
                    "versions.yml:md5,f833e7c777ef6137704988f225419557"
                ],
                "bed": [
                    [
                        {
                            "case": "finequagga",
                            "id": "ACC5963A1",
                            "sample": "ACC5963A1",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        },
                        "ACC5963A1_junction.bed.gz:md5,af36db589fe4bf79ab49944ef10d0891",
                        "ACC5963A1_junction.bed.gz.tbi:md5,30d9506483e03a1bc0fba299faba1a65"
                    ]
                ],
                "bw": [
                    [
                        {
                            "case": "finequagga",
                            "id": "ACC5963A1",
                            "sample": "ACC5963A1",
                            "strandedness": "reverse",
                            "sex": "NA",
                            "single_end": false,
                            "fq_pairs": 1,
                            "is_fastq": true
                        },
                        "ACC5963A1.bw:md5,bd693d63017f2d313aa50d8befdf339a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5abec6f51f30415625cf8ce28c698a72",
                    "versions.yml:md5,f4eb74ccf752488d150f4be9e1eb2142",
                    "versions.yml:md5,f833e7c777ef6137704988f225419557"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.5"
        },
        "timestamp": "2025-07-29T15:48:58.964592771"
    }
}